perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -1722,16 +2529,83 @@ public List<RichGroup> call(ApiCaller ac, Deserializer parms) throws PerunExcept
 	 * @param attrNames List<String> if attrNames is null method will return RichGroups containing all attributes
+	 * @param roles list of selected roles (if empty, then return groups by all roles)
+	 * @param types list of selected types of roles (if empty, then return by roles of all types)
 	 * @return List<RichGroup> RichGroups containing selected attributes
 	 */
+	/*#
+	 * Returns full list of member's RichGroups containing all attributes.
+	 *
+	 *
+	 * "members" group is not included!
+	 *
+	 * Supported are attributes from these namespaces:
+	 *  - group
+	 *  - member-group
+	 *
+	 * @throw MemberNotExistsException When the member doesn't exist
+	 *
+	 * @param member int <code>id</code> of member
+	 * @return List<RichGroup> RichGroups containing all attributes
+	 */
+	/*#
+	 * Returns full list of member's RichGroups filtered by user's role, containing all attributes.
+	 *
+	 *
+	 * "members" group is not included!
+	 *
+	 * Supported are attributes from these namespaces:
+	 *  - group
+	 *  - member-group
+	 *
+	 * @throw MemberNotExistsException When the member doesn't exist
+	 *
+	 * @param member int <code>id</code> of member
+	 * @param roles list of selected roles (if empty, then return groups by all roles)
+	 * @return List<RichGroup> RichGroups containing all attributes
+	 */
+	/*#
+	 * Returns full list of member's RichGroups filtered by user's role assignment type, containing all attributes.
+	 *
+	 *
+	 * "members" group is not included!
+	 *
+	 * Supported are attributes from these namespaces:
+	 *  - group
+	 *  - member-group
+	 *
+	 * @throw MemberNotExistsException When the member doesn't exist
+	 *
+	 * @param member int <code>id</code> of member
+	 * @param types list of selected types of roles (if empty, then return by roles of all types)
+	 * @return List<RichGroup> RichGroups containing all attributes
+	 */
+	/*#
+	 * Returns full list of member's RichGroups filtered by user's role and its assignment type, containing all attributes.
+	 *
+	 *
+	 * "members" group is not included!
+	 *
+	 * Supported are attributes from these namespaces:
+	 *  - group
+	 *  - member-group
+	 *
+	 * @throw MemberNotExistsException When the member doesn't exist
+	 *
+	 * @param member int <code>id</code> of member
+	 * @param roles list of selected roles (if empty, then return groups by all roles)
+	 * @param types list of selected types of roles (if empty, then return by roles of all types)
+	 * @return List<RichGroup> RichGroups containing all attributes
+	 */
 	getMemberRichGroupsWithAttributesByNames {
 
 		@Override
 		public List<RichGroup> call(ApiCaller ac, Deserializer parms) throws PerunException {
-
 			List<RichGroup> listOfRichGroups = ac.getGroupsManager().getMemberRichGroupsWithAttributesByNames(ac.getSession(),
-				ac.getMemberById(parms.readInt("member")),
-				parms.readList("attrNames", String.class));
+					ac.getMemberById(parms.readInt("member")),
+					parms.readList("attrNames", String.class),
```

Comment:
You're defining here in javadoc for `getMemberRichGroupsWithAttributesByNames` method the `attrNames` parameter as optional, which is not a problem if we want it now (because before this method had required parameter attrNames) - but if we want it, then there should be a null replacement for this param if it is not presented in `parms`.